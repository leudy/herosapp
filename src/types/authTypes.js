export const types = {
  LOGIN: " [AUTH] LOGIN",
  LOGOUT: " [AUTH] LOGOUT",
};
