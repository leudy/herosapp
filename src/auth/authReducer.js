import { types } from "../types/authTypes";

export const authReducer = (state = {}, action) => {


  
  switch (action.type) {
    case types.LOGIN: {
      const Data = { ...action.payload, logged: true };
      localStorage.setItem("user", JSON.stringify(Data));
      return Data;
    }
    case types.LOGOUT: {
      localStorage.removeItem("user");
      return { logged: false };
    }

    default:
      return state;
  }
};
