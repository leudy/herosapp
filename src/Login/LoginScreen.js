import React, { useContext } from "react";
import { types } from "../types/authTypes";
import { AuthContext } from "../auth/AuthContext";

export const LoginScreen = ({ history }) => {
  const { dispatch } = useContext(AuthContext);
  const handleLogin = () => {
    var root = localStorage.getItem("lastpath") || "/";
    if (root === "/") {
      localStorage.removeItem("param");
    }
    if (root !== "/" || root.length > 1) {
      if(localStorage.getItem('param')!==null)
      {
        root = root + localStorage.getItem("param") || "";
      }
    }
    dispatch({
      type: types.LOGIN,
      payload: { name: "Leudi Rosario, Reactjs" },
    });
    history.replace(root);
  };

  return (
    <div className="container">
      <h1>Login</h1>
      <p></p>
      <hr />
      <div className="col-md-12">
        <div className="col-md-4"></div>
        <div className="col-md-4">
          <button
            type="button"
            className="btn btn-outline-primary btn-block"
            onClick={handleLogin}
          >
            Entrar
          </button>
        </div>
        <div className="col-md-4"></div>
      </div>
    </div>
  );
};
