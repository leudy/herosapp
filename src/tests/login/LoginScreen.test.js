import React from 'react';
import "@testing-library/jest-dom"
import { mount,  configure} from "enzyme";    
import Adapter from "enzyme-adapter-react-16"

import { LoginScreen } from "../../Login/LoginScreen"
import { AuthContext } from '../../auth/AuthContext';
import {types } from "../../types/authTypes"


configure({adapter: new Adapter()});


describe('Prueba de Login Screen <LoginScreen />', () => {
   // creating mocks for test
    const ContextValues = {
        dispatch:jest.fn(),
        payload:{
            user:{name:'REACJS'}
        }
    }
    const HistoryMock = {
        replace: jest.fn()
    }
            const wrapper = mount(
               <AuthContext.Provider value={ContextValues}>
                    <LoginScreen history={HistoryMock}></LoginScreen>
               </AuthContext.Provider>);

  const handleClick = wrapper.find('button').prop('onClick');
  
  
  test('Debe mostrarse correctamente', () => {
      // make snapshot for the display
     console.log(wrapper.html())
     expect(wrapper).toMatchSnapshot();
  })
   
  test('debe realizar el dispatch ', () => {

    
    handleClick();
    expect(ContextValues.dispatch).toHaveBeenCalledWith({
        type: types.LOGIN,
        payload: { name: "Leudi Rosario, Reactjs" },
      });
      
      //lastpath
    
  })
  test('Probando el localstorage', () => {
    expect(HistoryMock.replace).toHaveBeenCalledWith('/')
    localStorage.setItem('lastpath','/dc')
    handleClick();
    expect(HistoryMock.replace).toHaveBeenCalledWith('/dc')

  })
  
  




})
