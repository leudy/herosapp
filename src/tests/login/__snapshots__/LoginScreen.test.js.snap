// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de Login Screen <LoginScreen /> Debe mostrarse correctamente 1`] = `ReactWrapper {}`;
